### PR TITLE
Add user group event relationships to db schema

### PR DESCRIPTION
**Pull request by andrelandgraf**
Issue: N/A

## Summary (1-2 sentences)

This PR adds relationship models for user-group and user-event relationships to the `prisma.schema` file.

## Details (reason and description of the changes)

We have the following product requirements:

- A user should be able to create/manage groups
- A user should be able to create/edit events
- A user should be able to join groups
- A user should be able to RSVP/join events

This can be translated into relationships between the user, group, and event database tables.

### Many-to-many user group relationship:

- A user can have many groups.
- A group can have many users.
- A user in a group has a role, like admin or member.

### Many-to-many user event relationship:

- A user can RSVP to many events.
- An event can have many users.

### Implementation details (how was this change implemented?) (optional)

Screenshot 2023-07-24 at 7 30 04 PM

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -11,11 +11,19 @@ datasource db {
   relationMode = "prisma"
 }
 
+enum Role {
```
Comment (andrelandgraf) [inline prisma/schema.prisma:14]: An enum enforces the value to be one of the different options. Inserting any other value will result in an error.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -25,22 +33,30 @@ model Password {
...
   events      Event[]
+  users       User[]
+}
+
+model User_Group {
+  userId  String
+  groupId String
+  role    Role
+
+  @@id([userId, groupId])
```
Comment (andrelandgraf) [inline prisma/schema.prisma:49]: The primary key of the `User_Group` table is defined by the combination of `userId` and `groupId`. This means each `userId`-`groupId` combination has to be unique.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -25,22 +33,30 @@ model Password {
...
   date        DateTime
   description String
-  group       Group @relation(fields: [groupId], references: [id])
-  groupId     String @unique
-}
+  group       Group    @relation(fields: [groupId], references: [id])
+  groupId     String
+  users       User[]
 
+  @@index([groupId], name: "groupId")
```
Comment (andrelandgraf) [inline prisma/schema.prisma:61]: Prisma complained that there is no index on the `groupId` field. An index improves performance when querying for the indexed field. This is added to add an index for the `groupId` field as recommended by Prisma.

File: prisma/schema.prisma [hunk 4]
```diff
@@ -25,22 +33,30 @@ model Password {
...
 }
 
 model Event {
-  id          String @id @default(uuid())
+  id          String   @id @default(uuid())
   name        String
   date        DateTime
   description String
-  group       Group @relation(fields: [groupId], references: [id])
-  groupId     String @unique
```
Comment (andrelandgraf) [inline prisma/schema.prisma:44]: A group can have many events, hence the `groupId` foreign key in the `Event` model cannot be `@unique`.

**Review by codemechanica — APPROVED** [review]

**Review by tonydangblog — APPROVED** [review]

**Pull request merged**